src/pages/Profile.js

```diff
@@ -58,6 +60,120 @@ const styles = {
 }
 
 
+const NihLink = ({ nihToken }) => {
+  /*
+   * Hooks
+   */
+  const [{ linkedNihUsername, linkExpireTime, datasetPermissions }, setNihStatus] = useState({})
+  const [loading, setLoading] = useState(false)
+  const [linking, setLinking] = useState(false)
+  const signal = useCancellation()
+
+  Utils.useOnMount(() => {
+    const { User } = Ajax(signal)
+
+    const linkNihAccount = _.flow(
+      withErrorReporting('Error linking NIH account'),
+      Utils.withBusyState(setLinking)
+    )(async nihToken => {
+      setNihStatus(await User.linkNihAccount(nihToken))
+    })
+
+    const loadNihStatus = _.flow(
+      withErrorReporting('Error loading NIH account status'),
+      Utils.withBusyState(setLoading)
+    )(async () => {
+      try {
+        setNihStatus(await User.getNihStatus())
+      } catch (error) {
+        if (error.status === 404) setNihStatus({})
+        else throw error
+      }
+    })
+
+    if (nihToken) {
+      // Clear the query string, but use replace so the back button doesn't take the user back to the token
+      Nav.history.replace({ search: '' })
+      linkNihAccount(nihToken)
+    } else {
+      loadNihStatus()
+    }
+  })
+
+
+  /*
+   * Render helpers
+   */
+  const makeAccountLinkLink = label => {
+    const nihRedirectUrl = `${window.location.protocol}//${window.location.host}/#profile?nih-username-token={token}`
```

Comment:
`window.location.origin` should include both the protocol and the host. Also, for the path portion it would be preferable to use `Nav.getLink` since this is one of our own URLs.